boa/src/syntax/parser/statement/return_stm/mod.rs

```diff
@@ -51,13 +51,13 @@ impl TokenParser for ReturnStatement {
                 _ => {}
             }
 
-            return Ok(Node::Return(None));
+            return Ok(Return::new::<Node, Option<_>>(None).into());
         }
 
         let expr = Expression::new(true, self.allow_yield, self.allow_await).parse(cursor)?;
 
         cursor.expect_semicolon(false, "return statement")?;
 
-        Ok(Node::return_node(expr))
+        Ok(Return::new(expr).into())
```

Comment:
This should also return a `Return` instead of a `Node`.

Reply:
Done (also getting the other areas which have similar problems)